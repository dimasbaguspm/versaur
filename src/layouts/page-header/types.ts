import type { TextProps } from '@/primitive'
import type { ReactNode, HTMLAttributes } from 'react'

/**
 * Props for PageHeader main container
 */
export interface PageHeaderProps extends HTMLAttributes<HTMLElement> {
  children?: ReactNode
}

/**
 * Props for PageHeaderTop - main header area (breadcrumbs, title, actions)
 */
export interface PageHeaderTopProps extends HTMLAttributes<HTMLDivElement> {
  children: ReactNode
}

/**
 * Props for PageHeaderBreadcrumbs - breadcrumbs section
 */
export interface PageHeaderBreadcrumbsProps
  extends HTMLAttributes<HTMLDivElement> {
  children: ReactNode
}

/**
 * Props for PageHeaderContent - title and subtitle area
 */
export interface PageHeaderContentProps extends HTMLAttributes<HTMLDivElement> {
  children: ReactNode
}

/**
 * Props for PageHeaderTitle - main title
 */
export type PageHeaderTitleProps = TextProps

/**
 * Props for PageHeaderSubtitle - subtitle/description
 */
export type PageHeaderSubtitleProps = TextProps

/**
 * Props for PageHeaderBadges - badges section
 */
export interface PageHeaderBadgesProps extends HTMLAttributes<HTMLDivElement> {
  children: ReactNode
}

/**
 * Props for PageHeaderActions - actions section (buttons, button groups)
 */
export interface PageHeaderActionsProps extends HTMLAttributes<HTMLDivElement> {
  children: ReactNode
}

/**
 * Props for PageHeaderBottom - bottom section (tabs, filters, etc.)
 */
export interface PageHeaderBottomProps extends HTMLAttributes<HTMLDivElement> {
  children: ReactNode
}
