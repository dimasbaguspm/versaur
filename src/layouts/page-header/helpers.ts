import { cva } from 'class-variance-authority'

/**
 * PageHeader root styles
 */
export const pageHeaderVariants = cva('w-full mt-4')

/**
 * PageHeaderTop styles - main header area
 * Mobile-first responsive: stacks vertically on mobile, horizontal on desktop
 */
export const pageHeaderTopVariants = cva(
  'flex flex-col sm:flex-row sm:items-start sm:justify-between gap-4 px-4 sm:px-6 mb-4'
)

/**
 * PageHeaderBreadcrumbs styles
 * Responsive padding for mobile
 */
export const pageHeaderBreadcrumbsVariants = cva('mb-4 px-4 sm:px-6')

/**
 * PageHeaderContent styles - title and subtitle area
 * Full width on mobile, flexible on desktop
 */
export const pageHeaderContentVariants = cva(
  'flex-1 min-w-0 flex flex-col gap-2 w-full sm:w-auto'
)

/**
 * PageHeaderBadges styles
 * Responsive wrapping for mobile
 */
export const pageHeaderBadgesVariants = cva('flex items-center gap-2 flex-wrap')

/**
 * PageHeaderActions styles
 * Stacks vertically on mobile, horizontal on desktop
 * Full width buttons on mobile
 */
export const pageHeaderActionsVariants = cva(
  'flex flex-col sm:flex-row gap-2 sm:gap-3 w-full sm:w-auto sm:ml-auto sm:shrink-0 [&>*]:w-full sm:[&>*]:w-auto'
)

/**
 * PageHeaderBottom styles - bottom section for tabs
 * Responsive padding and overflow handling
 */
export const pageHeaderBottomVariants = cva(
  'w-full flex items-center px-4 sm:px-6 py-0 overflow-x-auto'
)
