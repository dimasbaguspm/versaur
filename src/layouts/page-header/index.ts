export { PageHeader } from './page-header'
export * from './types'
