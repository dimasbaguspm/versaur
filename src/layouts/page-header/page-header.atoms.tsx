import { forwardRef } from 'react'
import { cn } from '@/utils/cn'
import type {
  PageHeaderTopProps,
  PageHeaderBreadcrumbsProps,
  PageHeaderContentProps,
  PageHeaderTitleProps,
  PageHeaderSubtitleProps,
  PageHeaderBadgesProps,
  PageHeaderActionsProps,
  PageHeaderBottomProps,
} from './types'
import {
  pageHeaderTopVariants,
  pageHeaderBreadcrumbsVariants,
  pageHeaderContentVariants,
  pageHeaderBadgesVariants,
  pageHeaderActionsVariants,
  pageHeaderBottomVariants,
} from './helpers'
import { Text } from '@/primitive'

/**
 * PageHeaderTop - main header area containing breadcrumbs, content, and actions
 */
export const PageHeaderTop = forwardRef<HTMLDivElement, PageHeaderTopProps>(
  ({ children, className, ...props }, ref) => (
    <div
      ref={ref}
      className={cn(pageHeaderTopVariants(), className)}
      data-versaur-page-header-top
      {...props}
    >
      {children}
    </div>
  )
)

/**
 * PageHeaderBreadcrumbs - breadcrumbs section
 */
export const PageHeaderBreadcrumbs = forwardRef<
  HTMLDivElement,
  PageHeaderBreadcrumbsProps
>(({ children, className, ...props }, ref) => (
  <div
    ref={ref}
    className={cn(pageHeaderBreadcrumbsVariants(), className)}
    data-versaur-page-header-breadcrumbs
    {...props}
  >
    {children}
  </div>
))

/**
 * PageHeaderContent - content area for title, subtitle, and badges
 */
export const PageHeaderContent = forwardRef<
  HTMLDivElement,
  PageHeaderContentProps
>(({ children, className, ...props }, ref) => (
  <div
    ref={ref}
    className={cn(pageHeaderContentVariants(), className)}
    data-versaur-page-header-content
    {...props}
  >
    {children}
  </div>
))

/**
 * PageHeaderTitle - main title heading
 * Responsive font size: smaller on mobile, larger on desktop
 */
export const PageHeaderTitle = forwardRef<
  HTMLHeadingElement,
  PageHeaderTitleProps
>((props, ref) => (
  <Text
    {...props}
    as='h1'
    fontSize='xl'
    fontWeight='semibold'
    ellipsis
    ref={ref}
  />
))

/**
 * PageHeaderSubtitle - subtitle/description text
 * Responsive font size and line clamping for mobile
 */
export const PageHeaderSubtitle = forwardRef<
  HTMLParagraphElement,
  PageHeaderSubtitleProps
>(({ className, ...props }, ref) => (
  <Text
    as='p'
    fontSize='sm'
    fontWeight='normal'
    clamp={2}
    ref={ref}
    className={cn('mb-2 sm:clamp-3', className)}
    {...props}
  />
))

/**
 * PageHeaderBadges - badges container
 */
export const PageHeaderBadges = forwardRef<
  HTMLDivElement,
  PageHeaderBadgesProps
>(({ children, className, ...props }, ref) => (
  <div
    ref={ref}
    className={cn(pageHeaderBadgesVariants(), className)}
    data-versaur-page-header-badges
    {...props}
  >
    {children}
  </div>
))

/**
 * PageHeaderActions - actions section for buttons and button groups
 */
export const PageHeaderActions = forwardRef<
  HTMLDivElement,
  PageHeaderActionsProps
>(({ children, className, ...props }, ref) => (
  <div
    ref={ref}
    className={cn(pageHeaderActionsVariants(), className)}
    data-versaur-page-header-actions
    {...props}
  >
    {children}
  </div>
))

/**
 * PageHeaderBottom - bottom section for tabs and filters
 */
export const PageHeaderBottom = forwardRef<
  HTMLDivElement,
  PageHeaderBottomProps
>(({ children, className, ...props }, ref) => (
  <div
    ref={ref}
    className={cn(pageHeaderBottomVariants(), className)}
    data-versaur-page-header-bottom
    {...props}
  >
    {children}
  </div>
))
