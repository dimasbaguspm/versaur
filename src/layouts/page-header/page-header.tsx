import { forwardRef, type ReactNode } from 'react'
import { cn } from '@/utils/cn'
import { pageHeaderOuterVariants, pageHeaderInnerVariants } from './helpers'
import type { PageHeaderProps } from './types'
import {
  PageHeaderTop,
  PageHeaderBreadcrumbs,
  PageHeaderContent,
  PageHeaderTitle,
  PageHeaderSubtitle,
  PageHeaderBadges,
  PageHeaderActions,
  PageHeaderBottom,
  PageHeaderMobileActions,
} from './page-header.atoms'

/**
 * PageHeader - Compound layout component for page headers
 *
 * Provides a flexible layout system for page headers that can include:
 * - Breadcrumbs navigation
 * - Title and subtitle
 * - Status badges
 * - Action buttons and button groups
 * - Bottom tabs or filters
 */

interface PageHeaderRootProps extends Omit<PageHeaderProps, 'title'> {
  /**
   * Simplified API: Page title
   */
  title: ReactNode
  /**
   * Simplified API: Page subtitle/description
   */
  subtitle?: ReactNode
  /**
   * Simplified API: Breadcrumbs navigation component
   */
  breadcrumbs?: ReactNode
  /**
   * Simplified API: Status badges
   */
  badges?: ReactNode
  /**
   * Simplified API: Action buttons/button groups
   */
  actions?: ReactNode

  /**
   * Simplified API: Mobile action buttons/button groups
   */
  mobileActions?: ReactNode
  /**
   * Simplified API: Bottom content (tabs, filters)
   */
  tabs?: ReactNode
}

const PageHeaderRoot = forwardRef<HTMLElement, PageHeaderRootProps>(
  (
    {
      children,
      className,
      size = 'fluid',
      backgroundColor = 'white',
      title,
      subtitle,
      breadcrumbs,
      badges,
      actions,
      mobileActions,
      tabs,
      ...props
    },
    ref
  ) => {
    const hasBottomContent = tabs

    return (
      <header
        ref={ref}
        className={cn(pageHeaderOuterVariants({ backgroundColor }))}
        role='banner'
        {...props}
      >
        <div className={cn(pageHeaderInnerVariants({ size }), className)}>
          {breadcrumbs && (
            <PageHeaderBreadcrumbs size={size}>
              {breadcrumbs}
            </PageHeaderBreadcrumbs>
          )}

          <PageHeaderTop size={size}>
            <PageHeaderContent>
              <div className='flex items-center justify-between'>
                <PageHeaderTitle>{title}</PageHeaderTitle>
                {mobileActions && (
                  <PageHeaderMobileActions>
                    {mobileActions}
                  </PageHeaderMobileActions>
                )}
              </div>
              {subtitle && <PageHeaderSubtitle>{subtitle}</PageHeaderSubtitle>}
              {badges && <PageHeaderBadges>{badges}</PageHeaderBadges>}
            </PageHeaderContent>
            {actions && <PageHeaderActions>{actions}</PageHeaderActions>}
          </PageHeaderTop>

          {hasBottomContent && <PageHeaderBottom>{tabs}</PageHeaderBottom>}
          {children}
        </div>
      </header>
    )
  }
)

export const PageHeader = PageHeaderRoot
