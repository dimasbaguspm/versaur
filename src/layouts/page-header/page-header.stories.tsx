/**
 * PageHeader stories for Storybook
 * Group: Layout
 * Demonstrates PageHeader usage patterns with size variants
 * Icons from lucide-react (dev only)
 */
import { useState } from 'react'
import { ArrowDownIcon, HomeIcon, PlusIcon, UploadIcon } from 'lucide-react'
import { PageHeader } from './index'
import { Breadcrumbs } from '@/navigation/breadcrumbs'
import { Tabs } from '@/navigation/tabs'
import { Button } from '@/primitive/button'
import { Badge } from '@/primitive/badge'
import { ButtonGroup } from '@/layouts/button-group'
import { ButtonIcon, Icon } from '@/primitive'

export default {
  title: 'Layouts/PageHeader',
  component: PageHeader,
  parameters: {
    layout: 'fullscreen',
  },
}

/**
 * Default page header with title, subtitle, and actions
 * Demonstrates fluid size (full width) and white background
 */
export const Default = () => (
  <PageHeader
    title='Dashboard'
    subtitle='Welcome back, John'
    actions={
      <ButtonGroup>
        <Button variant='primary'>New Project</Button>
      </ButtonGroup>
    }
    mobileActions={
      <ButtonGroup>
        <ButtonIcon as={PlusIcon} variant='primary' aria-label='New Project' />
      </ButtonGroup>
    }
  />
)

/**
 * Complete page header with all elements including breadcrumbs, badges, and tabs
 * Demonstrates wide size (max-width container)
 */
export const Complete = () => {
  const [activeTab, setActiveTab] = useState('all')
  return (
    <PageHeader
      size='wide'
      title='User Management'
      subtitle='Manage user accounts and permissions across the organization'
      breadcrumbs={
        <Breadcrumbs>
          <Breadcrumbs.Item
            href='/'
            icon={<Icon as={HomeIcon} color='inherit' size='sm' />}
          >
            Home
          </Breadcrumbs.Item>
          <Breadcrumbs.Separator />
          <Breadcrumbs.Item href='/admin'>Admin</Breadcrumbs.Item>
          <Breadcrumbs.Separator />
          <Breadcrumbs.Item>Users</Breadcrumbs.Item>
        </Breadcrumbs>
      }
      badges={
        <>
          <Badge>Active</Badge>
          <Badge color='neutral'>Admin</Badge>
        </>
      }
      actions={
        <ButtonGroup>
          <Button variant='ghost'>
            <Icon as={UploadIcon} color='inherit' />
            Import
          </Button>
          <Button variant='outline'>
            <Icon as={ArrowDownIcon} color='inherit' />
            Export
          </Button>
          <Button variant='primary'>
            <Icon as={PlusIcon} color='inherit' />
            Add User
          </Button>
        </ButtonGroup>
      }
      mobileActions={
        <ButtonGroup>
          <ButtonIcon as={UploadIcon} variant='ghost' aria-label='Import' />
          <ButtonIcon
            as={ArrowDownIcon}
            variant='outline'
            aria-label='Export'
          />
          <ButtonIcon as={PlusIcon} variant='primary' aria-label='Add User' />
        </ButtonGroup>
      }
      tabs={
        <Tabs value={activeTab} onValueChange={setActiveTab}>
          <Tabs.Trigger value='all'>All Users</Tabs.Trigger>
          <Tabs.Trigger value='active'>Active</Tabs.Trigger>
          <Tabs.Trigger value='pending'>Pending</Tabs.Trigger>
          <Tabs.Trigger value='disabled'>Disabled</Tabs.Trigger>
        </Tabs>
      }
    />
  )
}

/**
 * Narrow header for focused content pages
 * Demonstrates narrow size (max-width 768px) centered layout
 */
export const Narrow = () => (
  <PageHeader
    size='narrow'
    title='Account Settings'
    subtitle='Manage your account preferences and security settings'
    breadcrumbs={
      <Breadcrumbs>
        <Breadcrumbs.Item href='/'>Home</Breadcrumbs.Item>
        <Breadcrumbs.Separator />
        <Breadcrumbs.Item>Settings</Breadcrumbs.Item>
      </Breadcrumbs>
    }
    actions={
      <ButtonGroup>
        <Button variant='outline'>Cancel</Button>
        <Button variant='primary'>Save Changes</Button>
      </ButtonGroup>
    }
    tabs={
      <Tabs value='all' onValueChange={() => {}}>
        <Tabs.Trigger value='all'>All Users</Tabs.Trigger>
        <Tabs.Trigger value='active'>Active</Tabs.Trigger>
        <Tabs.Trigger value='pending'>Pending</Tabs.Trigger>
        <Tabs.Trigger value='disabled'>Disabled</Tabs.Trigger>
      </Tabs>
    }
  />
)
