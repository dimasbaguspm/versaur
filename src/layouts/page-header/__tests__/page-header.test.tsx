import { render } from '@testing-library/react'
import { PageHeader } from '../index'
import { composeStories } from '@storybook/react'
import * as stories from '../page-header.stories'

const { Default, Complete, Narrow } = composeStories(stories)

describe('PageHeader', () => {
  it('renders correctly', () => {
    const { container } = render(<Default />)
    expect(container).toMatchSnapshot()
  })

  it('renders default layout with title and actions', () => {
    const { getByText } = render(<Default />)
    expect(getByText('Dashboard')).toBeInTheDocument()
    expect(getByText('Welcome back, John')).toBeInTheDocument()
    expect(getByText('New Project')).toBeInTheDocument()
  })

  it('renders complete layout with all elements', () => {
    const { getByText } = render(<Complete />)
    expect(getByText('User Management')).toBeInTheDocument()
    expect(getByText('Home')).toBeInTheDocument()
    expect(getByText('All Users')).toBeInTheDocument()
    expect(getByText('Add User')).toBeInTheDocument()
  })

  it('renders narrow layout with size prop', () => {
    const { getByText } = render(<Narrow />)
    expect(getByText('Account Settings')).toBeInTheDocument()
    expect(getByText('Settings')).toBeInTheDocument()
  })

  it('has correct semantic structure', () => {
    const { container } = render(<Default />)
    const header = container.querySelector('header[role="banner"]')
    expect(header).toBeInTheDocument()
  })

  it('applies size and backgroundColor props correctly', () => {
    const { container } = render(
      <PageHeader
        title='Test'
        size='wide'
        backgroundColor='gray'
        data-testid='header'
      />
    )
    const header = container.querySelector('header')
    expect(header).toHaveClass('bg-neutral')
  })

  it('renders both simplified props and children correctly', () => {
    const { getByText } = render(
      <PageHeader title='Prop Title'>
        <div>Custom Child Content</div>
      </PageHeader>
    )
    expect(getByText('Prop Title')).toBeInTheDocument()
    expect(getByText('Custom Child Content')).toBeInTheDocument()
  })
})
