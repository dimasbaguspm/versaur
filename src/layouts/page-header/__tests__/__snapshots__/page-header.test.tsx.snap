// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`PageHeader > renders complete layout with all elements 1`] = `
<div>
  <header
    class="w-full"
    role="banner"
  >
    <div
      class="mb-4 px-4 sm:px-6"
      data-versaur-page-header-breadcrumbs="true"
    >
      <nav
        aria-label="Breadcrumb"
      >
        <ol
          class="flex items-center gap-2 text-sm text-gray-500"
        >
          <li
            class="flex items-center"
          >
            <a
              class="inline-flex items-center gap-1 transition-colors hover:underline cursor-pointer"
              href="/"
              tabindex="0"
            >
              Home
            </a>
          </li>
          <svg
            aria-hidden="true"
            class="lucide lucide-chevron-right inline-flex items-center justify-center w-3 h-3"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="m9 18 6-6-6-6"
            />
          </svg>
          <li
            class="flex items-center"
          >
            <a
              class="inline-flex items-center gap-1 transition-colors hover:underline cursor-pointer"
              href="/admin"
              tabindex="0"
            >
              Admin
            </a>
          </li>
          <svg
            aria-hidden="true"
            class="lucide lucide-chevron-right inline-flex items-center justify-center w-3 h-3"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="m9 18 6-6-6-6"
            />
          </svg>
          <li
            class="flex items-center"
          >
            <a
              class="inline-flex items-center gap-1 transition-colors hover:underline cursor-pointer"
              tabindex="0"
            >
              Users
            </a>
          </li>
        </ol>
      </nav>
    </div>
    <div
      class="flex flex-col sm:flex-row sm:items-start sm:justify-between gap-4 px-4 sm:px-6 mb-4"
      data-versaur-page-header-top="true"
    >
      <div
        class="flex-1 min-w-0 flex flex-col gap-2 w-full sm:w-auto"
        data-versaur-page-header-content="true"
      >
        <h1
          class="text-ghost text-left truncate text-xl font-semibold"
        >
          User Management
        </h1>
        <p
          class="text-ghost text-left line-clamp-2 text-sm font-normal mb-2 sm:clamp-3"
        >
          Manage user accounts and permissions across the organization
        </p>
        <div
          class="flex items-center gap-2 flex-wrap"
          data-versaur-page-header-badges="true"
        >
          <span
            class="inline-flex items-center justify-center font-medium transition-all duration-200 select-none gap-1 min-h-[1.25rem] rounded-md text-xs px-1.5 py-0.5 h-4 bg-primary-light text-primary"
          >
            Active
          </span>
          <span
            class="inline-flex items-center justify-center font-medium transition-all duration-200 select-none gap-1 min-h-[1.25rem] rounded-md text-xs px-1.5 py-0.5 h-4 bg-neutral text-foreground"
          >
            Admin
          </span>
        </div>
      </div>
      <div
        class="flex flex-col sm:flex-row gap-2 sm:gap-3 w-full sm:w-auto sm:ml-auto sm:shrink-0 [&>*]:w-full sm:[&>*]:w-auto"
        data-versaur-page-header-actions="true"
      >
        <div
          class="flex flex-row justify-start gap-3"
          role="group"
        >
          <button
            aria-disabled="false"
            class="inline-flex items-center justify-center gap-2 rounded-md font-medium transition-all duration-200 focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-offset-2 disabled:opacity-50 disabled:pointer-events-none active:scale-[0.98] select-none cursor-pointer bg-white text-foreground hover:bg-ghost-soft focus-visible:ring-ghost-light focus-visible:ring-offset-white h-9 px-4 text-sm min-w-[2.5rem]"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="lucide lucide-upload inline-flex items-center justify-center w-5 h-5"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M12 3v12"
              />
              <path
                d="m17 8-5-5-5 5"
              />
              <path
                d="M21 15v4a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2v-4"
              />
            </svg>
            Import
          </button>
          <button
            aria-disabled="false"
            class="inline-flex items-center justify-center gap-2 rounded-md font-medium transition-all duration-200 focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-offset-2 disabled:opacity-50 disabled:pointer-events-none active:scale-[0.98] select-none cursor-pointer bg-secondary text-white hover:bg-secondary/90 focus-visible:ring-secondary-light focus-visible:ring-offset-white shadow-sm hover:shadow-md h-9 px-4 text-sm min-w-[2.5rem]"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="lucide lucide-arrow-down inline-flex items-center justify-center w-5 h-5"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M12 5v14"
              />
              <path
                d="m19 12-7 7-7-7"
              />
            </svg>
            Export
          </button>
          <button
            aria-disabled="false"
            class="inline-flex items-center justify-center gap-2 rounded-md font-medium transition-all duration-200 focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-offset-2 disabled:opacity-50 disabled:pointer-events-none active:scale-[0.98] select-none cursor-pointer bg-primary text-white hover:bg-primary/90 focus-visible:ring-primary-light focus-visible:ring-offset-white shadow-sm hover:shadow-md h-9 px-4 text-sm min-w-[2.5rem]"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="lucide lucide-plus inline-flex items-center justify-center w-5 h-5"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M5 12h14"
              />
              <path
                d="M12 5v14"
              />
            </svg>
            Add User
          </button>
        </div>
      </div>
    </div>
    <div
      class="w-full flex items-center px-4 sm:px-6 py-0 overflow-x-auto"
      data-versaur-page-header-bottom="true"
    >
      <div
        class="flex flex-row w-full overflow-x-auto whitespace-nowrap border-b border-border"
        role="tablist"
        style="position: relative;"
      >
        <button
          aria-controls="tabs-content-all"
          aria-selected="true"
          class="inline-flex items-center justify-center transition-all duration-200 focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-offset-2 disabled:opacity-50 disabled:pointer-events-none select-none px-4 py-2 text-sm relative font-medium bg-transparent border-b-2 text-primary border-primary"
          id="tabs-trigger-all"
          role="tab"
          tabindex="0"
          type="button"
        >
          All Users
        </button>
        <button
          aria-controls="tabs-content-active"
          aria-selected="false"
          class="inline-flex items-center justify-center transition-all duration-200 focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-offset-2 disabled:opacity-50 disabled:pointer-events-none select-none px-4 py-2 text-sm relative font-normal bg-transparent border-b-2 border-transparent text-foreground hover:text-primary"
          id="tabs-trigger-active"
          role="tab"
          tabindex="-1"
          type="button"
        >
          Active
        </button>
        <button
          aria-controls="tabs-content-pending"
          aria-selected="false"
          class="inline-flex items-center justify-center transition-all duration-200 focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-offset-2 disabled:opacity-50 disabled:pointer-events-none select-none px-4 py-2 text-sm relative font-normal bg-transparent border-b-2 border-transparent text-foreground hover:text-primary"
          id="tabs-trigger-pending"
          role="tab"
          tabindex="-1"
          type="button"
        >
          Pending
        </button>
        <button
          aria-controls="tabs-content-disabled"
          aria-selected="false"
          class="inline-flex items-center justify-center transition-all duration-200 focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-offset-2 disabled:opacity-50 disabled:pointer-events-none select-none px-4 py-2 text-sm relative font-normal bg-transparent border-b-2 border-transparent text-foreground hover:text-primary"
          id="tabs-trigger-disabled"
          role="tab"
          tabindex="-1"
          type="button"
        >
          Disabled
        </button>
        <div
          aria-hidden="true"
          class="absolute bottom-0 h-0.5 rounded transition-all duration-300 bg-primary"
          style="left: 0px; width: 0px; pointer-events: none;"
        />
      </div>
    </div>
  </header>
</div>
`;

exports[`PageHeader > renders correctly with simple props 1`] = `
<div>
  <header
    class="w-full"
    role="banner"
  >
    <div
      class="mb-4 px-4 sm:px-6"
      data-versaur-page-header-breadcrumbs="true"
    >
      <nav
        aria-label="Breadcrumb"
      >
        <ol
          class="flex items-center gap-2 text-sm text-gray-500"
        >
          <li
            class="flex items-center"
          >
            <a
              class="inline-flex items-center gap-1 transition-colors hover:underline cursor-pointer"
              href="/"
              tabindex="0"
            >
              <span
                class="mr-1 inline-flex"
              >
                <svg
                  aria-hidden="true"
                  class="lucide lucide-house inline-flex items-center justify-center w-4 h-4"
                  fill="none"
                  height="24"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M15 21v-8a1 1 0 0 0-1-1h-4a1 1 0 0 0-1 1v8"
                  />
                  <path
                    d="M3 10a2 2 0 0 1 .709-1.528l7-5.999a2 2 0 0 1 2.582 0l7 5.999A2 2 0 0 1 21 10v9a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2z"
                  />
                </svg>
              </span>
              Home
            </a>
          </li>
          <svg
            aria-hidden="true"
            class="lucide lucide-chevron-right inline-flex items-center justify-center w-3 h-3"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="m9 18 6-6-6-6"
            />
          </svg>
          <li
            class="flex items-center"
          >
            <a
              class="inline-flex items-center gap-1 transition-colors hover:underline cursor-pointer"
              href="/admin"
              tabindex="0"
            >
              Admin
            </a>
          </li>
          <svg
            aria-hidden="true"
            class="lucide lucide-chevron-right inline-flex items-center justify-center w-3 h-3"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="m9 18 6-6-6-6"
            />
          </svg>
          <li
            class="flex items-center"
          >
            <a
              class="inline-flex items-center gap-1 transition-colors hover:underline cursor-pointer"
              tabindex="0"
            >
              Users
            </a>
          </li>
        </ol>
      </nav>
    </div>
    <div
      class="flex flex-col sm:flex-row sm:items-start sm:justify-between gap-4 px-4 sm:px-6 mb-4"
      data-versaur-page-header-top="true"
    >
      <div
        class="flex-1 min-w-0 flex flex-col gap-2 w-full sm:w-auto"
        data-versaur-page-header-content="true"
      >
        <h1
          class="text-ghost text-left truncate text-xl font-semibold"
        >
          Dashboard
        </h1>
        <p
          class="text-ghost text-left line-clamp-2 text-sm font-normal mb-2 sm:clamp-3"
        >
          Welcome back, John
        </p>
      </div>
      <div
        class="flex flex-col sm:flex-row gap-2 sm:gap-3 w-full sm:w-auto sm:ml-auto sm:shrink-0 [&>*]:w-full sm:[&>*]:w-auto"
        data-versaur-page-header-actions="true"
      >
        <div
          class="flex flex-row justify-start gap-3"
          role="group"
        >
          <button
            aria-disabled="false"
            class="inline-flex items-center justify-center gap-2 rounded-md font-medium transition-all duration-200 focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-offset-2 disabled:opacity-50 disabled:pointer-events-none active:scale-[0.98] select-none cursor-pointer bg-primary text-white hover:bg-primary/90 focus-visible:ring-primary-light focus-visible:ring-offset-white shadow-sm hover:shadow-md h-9 px-4 text-sm min-w-[2.5rem]"
            type="button"
          >
            New Project
          </button>
        </div>
      </div>
    </div>
  </header>
</div>
`;

exports[`PageHeader > renders minimal layout 1`] = `
<div>
  <header
    class="w-full"
    role="banner"
  >
    <div
      class="mb-4 px-4 sm:px-6"
      data-versaur-page-header-breadcrumbs="true"
    >
      <nav
        aria-label="Breadcrumb"
      >
        <ol
          class="flex items-center gap-2 text-sm text-gray-500"
        >
          <li
            class="flex items-center"
          >
            <a
              class="inline-flex items-center gap-1 transition-colors hover:underline cursor-pointer"
              href="/"
              tabindex="0"
            >
              <span
                class="mr-1 inline-flex"
              >
                <svg
                  aria-hidden="true"
                  class="lucide lucide-house inline-flex items-center justify-center w-4 h-4"
                  fill="none"
                  height="24"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M15 21v-8a1 1 0 0 0-1-1h-4a1 1 0 0 0-1 1v8"
                  />
                  <path
                    d="M3 10a2 2 0 0 1 .709-1.528l7-5.999a2 2 0 0 1 2.582 0l7 5.999A2 2 0 0 1 21 10v9a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2z"
                  />
                </svg>
              </span>
              Home
            </a>
          </li>
          <svg
            aria-hidden="true"
            class="lucide lucide-chevron-right inline-flex items-center justify-center w-3 h-3"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="m9 18 6-6-6-6"
            />
          </svg>
          <li
            class="flex items-center"
          >
            <a
              class="inline-flex items-center gap-1 transition-colors hover:underline cursor-pointer"
              href="/admin"
              tabindex="0"
            >
              Admin
            </a>
          </li>
          <svg
            aria-hidden="true"
            class="lucide lucide-chevron-right inline-flex items-center justify-center w-3 h-3"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="m9 18 6-6-6-6"
            />
          </svg>
          <li
            class="flex items-center"
          >
            <a
              class="inline-flex items-center gap-1 transition-colors hover:underline cursor-pointer"
              tabindex="0"
            >
              Users
            </a>
          </li>
        </ol>
      </nav>
    </div>
    <div
      class="flex flex-col sm:flex-row sm:items-start sm:justify-between gap-4 px-4 sm:px-6 mb-4"
      data-versaur-page-header-top="true"
    >
      <div
        class="flex-1 min-w-0 flex flex-col gap-2 w-full sm:w-auto"
        data-versaur-page-header-content="true"
      >
        <h1
          class="text-ghost text-left truncate text-xl font-semibold"
        >
          Simple Page
        </h1>
      </div>
    </div>
  </header>
</div>
`;
